src/test/scala/org/broadinstitute/dsde/firecloud/webservice/WorkspaceApiServiceSpec.scala

```diff
@@ -1021,14 +1027,35 @@ class WorkspaceApiServiceSpec extends BaseServiceSpec with WorkspaceApiService w
       }
 
       "should 400 if a bagit request has an invalid format" in {
-        bagitService()
         stubRawlsService(HttpMethods.POST, s"$workspacesPath/entities/batchUpsert", NoContent)
         (Post(bagitImportPath, HttpEntity(MediaTypes.`application/json`, s"""{"bagitURL":"https://localhost:$bagitServerPort/both.zip", "format":"garbage" }"""))
           ~> dummyUserIdHeaders(dummyUserId)
           ~> sealRoute(workspaceRoutes)) ~> check {
           status should equal(BadRequest)
         }
       }
+
+      "should 400 if a bagit is empty" in {
+        stubRawlsService(HttpMethods.POST, s"$workspacesPath/entities/batchUpsert", NoContent)
+        (Post(bagitImportPath, HttpEntity(MediaTypes.`application/json`, s"""{"bagitURL":"https://localhost:$bagitServerPort/empty.zip", "format":"TSV" }"""))
+          ~> dummyUserIdHeaders(dummyUserId)
+          ~> sealRoute(workspaceRoutes)) ~> check {
+          status should equal(BadRequest)
+          val errRpt = responseAs[ErrorReport]
+          errRpt.message shouldBe s"BDBag has no entries."
+        }
+      }
+
+      "should 400 if a bagit is unzippable" in {
+        stubRawlsService(HttpMethods.POST, s"$workspacesPath/entities/batchUpsert", NoContent)
+        (Post(bagitImportPath, HttpEntity(MediaTypes.`application/json`, s"""{"bagitURL":"https://localhost:$bagitServerPort/notazip.zip", "format":"TSV" }"""))
+          ~> dummyUserIdHeaders(dummyUserId)
+          ~> sealRoute(workspaceRoutes)) ~> check {
+          status should equal(BadRequest)
+          val errRpt = responseAs[ErrorReport]
+          errRpt.message shouldBe s"Problem with BDBag: zip END header not found"
```

Comment:
I'm probably missing something…but how do we know to test for this particular error string? If it's external, would it always stay the same?

Reply:
the "zip END header not found" string comes from `java.util.zip.ZipEntry` - or one of the related classes in `java.util.zip`. Since it's part of Java itself, I don't expect it to change frequently or carelessly - but yeah, it's external